sdks/python/container/base_image_requirements_manual.txt

```diff
@@ -0,0 +1,41 @@
+###############################################################################
+#  Licensed to the Apache Software Foundation (ASF) under one
+#  or more contributor license agreements.  See the NOTICE file
+#  distributed with this work for additional information
+#  regarding copyright ownership.  The ASF licenses this file
+#  to you under the Apache License, Version 2.0 (the
+#  "License"); you may not use this file except in compliance
+#  with the License.  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+# limitations under the License.
+###############################################################################
+
+# This file has additional packages to install in Beam's containers.
+# Do not specify dependencies, which are already in Beam's setup.py.
+# See generate_pip_requirements_list.sh to find which extras (e.g. gcp) are used.
+# Do not specify a constrained version without a particular reason, so that
+# we can pick up recent versions when regenerating the requirements.
+# Consider constraining requirements of Beam itself when necessary.
+
+bs4  # Commonly used HTML processing tool.
+# Let's upgrade to Cython 3.0 in a controlled way, stay on 0.x versions for now.
+cython<1
+google-cloud-profiler
+google-python-cloud-debugger
+guppy3  # Memory profiler
+mmh3  # Optimzes execution of some Beam codepaths. TODO: Make it Beam's dependency.
+python-snappy  # Optimzes execution of some Beam codepaths.
+scipy
+scikit-learn
```

Comment:
I'm just curious why some of these packages like scikit-learn and tensorflow are specified as deps.  Seems like they shouldn't be.

Reply:
They are deemed to be commonly used; we don't have good visibility though how commonly they are used. For purposes of this PR, I am preserving the current status quo and just moving them into a separate file that has optional dependencies.